scripts/mixins/tconstruct.zs

```diff
@@ -0,0 +1,76 @@
+#loader mixin
+
+import native.net.minecraft.item.ItemStack;
+import native.net.minecraftforge.fluids.Fluid;
+import native.net.minecraft.nbt.NBTTagCompound;
+import native.slimeknights.tconstruct.library.smeltery.ICastingRecipe;
+import native.slimeknights.tconstruct.library.modifiers.ModifierAspect;
+import native.slimeknights.tconstruct.library.modifiers.ModifierNBT;
+import native.slimeknights.tconstruct.library.tools.ToolNBT;
+import native.slimeknights.tconstruct.library.utils.TagUtil;
+import native.slimeknights.tconstruct.tools.modifiers.ToolModifier;
+import mixin.CallbackInfo;
+
+#mixin Mixin
+#{targets: "slimeknights.tconstruct.smeltery.TinkerSmeltery"}
+zenClass MixinTinkerSmeltery {
+    #mixin Static
+    #mixin Redirect
+    #{
+    #    method: "registerToolpartMeltingCasting",
+    #    at: {
+    #        value: "INVOKE",
+    #        target: "Lslimeknights/tconstruct/library/TinkerRegistry;registerMelting(Lnet/minecraft/item/ItemStack;Lnet/minecraftforge/fluids/Fluid;I)V",
+    #        ordinal: 0
+    #    }
+    #}
+    function disableToolpartMelting(stack as ItemStack, fluid as Fluid, amount as int) as void {
+        // NO-OP
+    }
+
+    #mixin Static
+    #mixin Redirect
+    #{
+    #    method: "registerToolpartMeltingCasting",
+    #    at: {
+    #        value: "INVOKE",
+    #        target: "Lslimeknights/tconstruct/library/TinkerRegistry;registerTableCasting(Lslimeknights/tconstruct/library/smeltery/ICastingRecipe;)V",
+    #        ordinal: 1
+    #    }
+    #}
+    function disableCastRecipes(recipe as ICastingRecipe) as void {
+        // NO-OP
+    }
```

Comment:
I cant understand why, but this tweak not working. I can still see all the variations of tool part castings in HEI.

![](https://i.imgur.com/UCQv0Ud.png)

This is kinda weird, since i manually adding `stone_part => cast` recipes with [this CT script](https://github.com/friendlyhj/Enigmatica2Expert-Extended/blob/cbf43de37b4343dd2efa0c3a2d0587ab6a980ffb/scripts/mods/tconstruct/tconstruct.zs#L532-L537), but unable to see those recipes in HEI.

This *may be* issue with a mixin, but also may be issue with CT script.

Reply:
New change didnt changed wrong behaviour and casts are still available from all possible materials:
![](https://i.imgur.com/UPpTbWx.png)

Reply:
Issue is fixed.